jpa/eclipselink/src/main/java/org/jipijapa/eclipselink/JndiUtil.java

```diff
@@ -0,0 +1,108 @@
+import javax.transaction.TransactionSynchronizationRegistry;
+
+/**
+ * Helper class that centralizes the jndi lookup logic.
+ */
+public final class JndiUtil {
+
+    public static final JndiUtil SINGLETON = new JndiUtil();
+
+    /**
+     * Create a new JndiUtil.
+     */
+    private JndiUtil() {
+
+    }
+
+    /**
+     * Obtain the container transaction manager.
+     *
+     * @return The container transaction manager (e.g.
+     *         com.arjuna.ats.jbossatx.jta.TransactionManagerDelegate)
+     * @throws Exception
+     *             Unexpected error takes place during jndi lookup
+     */
+    public TransactionManager acquireTransactionManager() throws Exception {
+        try {
+            return InitialContext.doLookup(JndiConstants.JBOSS_TRANSACTION_MANAGER);
+        } catch (NamingException ex) {
+            String errMsg = createErrMsgForFailedJndiLookup(ex, JndiConstants.JBOSS_TRANSACTION_MANAGER);
+            throw new Exception(errMsg, ex);
+        }
+    }
+
+    /**
+     * Obtain the transaction synchronization registry from the container via
+     * jndi lookup.
+     *
+     * @return The transaction synchronization registry (e.g.
+     *         org.jboss.as.txn.service.internal.tsr.TransactionSynchronizationRegistryWrapper)
+     *
+     * @throws Exception
+     *             A checked exception is fired whenever an unexpected error
+     *             takes place (e.g. the jndi lookup of
+     *             {@link JndiConstants#JTA_TRANSACTION_SYNCHRONIZATION_REGISTRY}
+     *             fails)
+     */
+    public TransactionSynchronizationRegistry acquireTransactionSynchronizationRegistry() throws Exception {
+        try {
+            return InitialContext.doLookup(JndiConstants.JTA_TRANSACTION_SYNCHRONIZATION_REGISTRY);
+        } catch (NamingException ex) {
+            // (a) Build an error message
+            // note: in this case we do not log the erro - since pump up an
+            // exception
+            // there is no need to do log spamming
+            String errMsg = createErrMsgForFailedJndiLookup(ex, JndiConstants.JTA_TRANSACTION_SYNCHRONIZATION_REGISTRY);
+
+            // (b) Attempt break the execution logic accessing this resource is
+            // critical.
+            throw new Exception(errMsg, ex);
+        }
+    }
+
+    /**
+     * Build an error message giving an indication that a resource expected to
+     * exist in the container and be accessible via jndi lookup was not found.
+     *
+     * @param namingException
+     *            A jndi lookup error
+     * @param jndiLookupName
+     *            The jndi name that could not be looked up
+     * @return An indicative error message
+     */
+    private String createErrMsgForFailedJndiLookup(NamingException namingException, String jndiLookupName) {
+        return String.format(
+                "Unexpected error took place while attempting to lookup the container TransactionSynchronizationRegistry. %n"
```

Comment:
Please also move this error to a BasicLogger, similar to what is suggested above.